src/services/dependency-resolving.ts

```diff
@@ -150,13 +156,18 @@ export function makeResolveDependency(
       if (!isProcessed) {
         // add entry to processed list
         processedList.push(entry);
-        const isInternal = isInternalPackage(entryName);
+        const isInternalResult = await checkIsBuiltInPackage(
+          entryName,
+          entryVersion
+        ).promise;
+        if (isInternalResult.isErr()) return isInternalResult;
+        const isInternal = isInternalResult.value;
         const isSelf = entryName === name;
 
         if (isInternal) {
           depsValid.push({
             name: entryName,
-            version: latestVersion,
+            version: entryVersion,
```

Comment:
Because the `entryVersion` is considered a valid dependency, it bypassed the version check. This should resolve #133 and related issues.